helper/util.go

```diff
@@ -798,3 +828,61 @@ func FetchFromAPI(cliCtx cliContext.CLIContext, URL string) (result rest.Respons
 	Logger.Debug("Error while fetching data from URL", "status", resp.StatusCode, "URL", URL)
 	return result, fmt.Errorf("Error while fetching data from url: %v, status: %v", URL, resp.StatusCode)
 }
+
+// EnrichWithGas calculates the gas estimate that would be consumed by the
+// transaction and set the transaction's respective value accordingly.
+func EnrichWithGas(txBldr authTypes.TxBuilder, cliCtx context.CLIContext, msgs []sdk.Msg) (authTypes.TxBuilder, error) {
+	_, adjusted, err := simulateMsgs(txBldr, cliCtx, msgs)
+	if err != nil {
+		return txBldr, err
+	}
+
+	return txBldr.WithGas(adjusted), nil
+}
+
+// nolint
+// SimulateMsgs simulates the transaction and returns the gas estimate and the adjusted value.
+func simulateMsgs(txBldr authTypes.TxBuilder, cliCtx context.CLIContext, msgs []sdk.Msg) (estimated, adjusted uint64, err error) {
+	txBytes, err := txBldr.BuildTxForSim(msgs)
+	if err != nil {
+		return
+	}
+
+	return CalculateGas(cliCtx.QueryWithData, cliCtx.Codec, txBytes, txBldr.GasAdjustment())
+}
+
+// CalculateGas simulates the execution of a transaction and returns
+// both the estimate obtained by the query and the adjusted amount.
+func CalculateGas(
+	queryFunc func(string, []byte) ([]byte, int64, error), cdc *codec.Codec,
+	txBytes []byte, adjustment float64,
+) (estimate, adjusted uint64, err error) {
+
+	// run a simulation (via /app/simulate query) to
+	// estimate gas and update TxBuilder accordingly
+	rawRes, _, err := queryFunc("/app/simulate", txBytes)
```

Comment:
if gas is more than 1m, dont send with adjustment value, send max of 1m gas limit